### PR TITLE
Use a fixed number of omp threads for unit tests on builders

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Avoids thread contention on the builders by settings a maximum number of the OpenMP threads that are used for the unit tests on the builders

**To test:**

* All test jobs should complete.

*There is no associated issue.*

*This does not require release notes* because **it is not a user-facing change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by SimonHeybrock — CHANGES_REQUESTED** [review]
What about performance tests? We should probably make sure those still run with the old/default setting? I cannot tell wether the change below also affects those?

**Comment by martyngigg** [comment]
It won't affect those as they have a different script, currently stuck in Jenkins, to run them. I have checked and we remove any `~/.mantid/Mantid.user.properties` before running the tests so it should be unchanged.

**Review by SimonHeybrock — APPROVED** [review]

**Comment by martyngigg** [comment]
It just occurred to me though that this will affect the documentation tests that are run sequentially so I may remove the file for those tests.

File: buildconfig/Jenkins/buildscript [hunk 1]
```diff
@@ -353,6 +354,8 @@ fi
 # User Documentation
 ###############################################################################
 if [[ ${DO_DOCTESTS_USER} == true ]]; then
+  # use default configuration
+  rm -f $userprops_file
```
Comment (SimonHeybrock) [inline buildconfig/Jenkins/buildscript]: Is it a problem that it is now missing, as apposed to empty, before removing line 346?

Comment (martyngigg) [reply buildconfig/Jenkins/buildscript]: It shouldn't be as the doc tests spin up MantidPlot and that has a single initialization point for mantid so there won't be a race-condition with creating the file.

File: buildconfig/Jenkins/buildscript [hunk 2]
```diff
@@ -342,7 +342,9 @@ fi
 userconfig_dir=$HOME/.mantid
 rm -fr $userconfig_dir
 mkdir -p $userconfig_dir
-touch $userconfig_dir/Mantid.user.properties
+# use a fixed number of openmp threads to avoid overloading the system
+userprops_file=$userconfig_dir/Mantid.user.properties
+echo MultiThreaded.MaxCores=2 > $userprops_file
```
Comment (peterfpeterson) [inline buildconfig/Jenkins/buildscript]: I thought the [recommendation](https://github.com/DMSC-Instrument-Data/documents/blob/eef67316a6c88d66cc1873ce2431c5646045a2d2/investigations/Ctest_and_threads/using_threads_with_ctest.md) was to set this to be `${BUILD_THREADS}/4`. The evaluation would be
```shell
echo "MultiThreaded.MaxCores=$((${BUILD_THREADS} / 4))"
```

**Review by peterfpeterson — CHANGES_REQUESTED** [review]

File: buildconfig/Jenkins/buildscript (hunk 2 above)
Comment (martyngigg) [reply buildconfig/Jenkins/buildscript]: I was going with the solution for maximum performance: https://github.com/DMSC-Instrument-Data/documents/blob/eef67316a6c88d66cc1873ce2431c5646045a2d2/investigations/Ctest_and_threads/using_threads_with_ctest.md#maximum-performance

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**